### DISCUSSION TITLE
Techstack behind virtuoso.

### DISCUSSION THREAD

**Discussion by wibed** [petyosi/react-virtuoso#824] [General]
what kind of virtualization library does virtuoso base its codebase on?

or is it event developer completely from scracth

**Comment by petyosi** [comment]
React Virtuoso is the virtualization library itself. I has zero dependencies ([package.json](https://github.com/petyosi/react-virtuoso/blob/master/package.json)).

**Reply by wibed** [reply]
looking back, 
would you prefer to have implemented an existing virtualization library?

**Reply by petyosi** [reply]
Not sure what you mean by an existing virtualization library - can you give some examples of such libraries?

**Reply by wibed** [reply]
react-virtualized
react-window
to name two idk if there are other virtualization libraries, except yours of course

**Reply by petyosi** [reply]
No, those can't be used to implement what virtuoso does.

**Reply by wibed** [reply]
could you enlighten me. how do they differ?
is your library the only library with virtualization?

**Reply by petyosi** [reply]
The main (and most critical) difference is how the libraries approach each item height, which is the main problem of the space. Most (if not all) alternatives choose to offload that to the integrator. Virtuoso handles this internally by default.

**Reply by wibed** [reply]
by integrator do you mean a browser api?

**Reply by wibed** [reply]
just to give some background,,,
id like to add virtualization to a scheduling component as dnd capabilities drag performance down the more events one loads simultaneously.

impressed by your implementation i started to research a bit and here we are.
